### PR TITLE
feat: add multiple policy support

### PR DESCRIPTION
**Pull request by tangyang9464**
Signed-off-by: tangyang9464 <tangyang9464@163.com>

fix: #835

### CODE REVIEW

**Review by sagilio — CHANGES_REQUESTED** [review]
I think it better to use a new API or parameter to support multiple types.

Casbin.NET plans to use `EnforceContext` to support multiple types or custom types. https://github.com/casbin/Casbin.NET/blob/8a3f1fd2cdbc176ec018d9cec263fc51853ad1b8/NetCasbin/Extensions/Enforcer/EnforcerExtension.cs#L398-L406

File: enforcer.go [hunk 1]
```diff
@@ -455,9 +457,32 @@ func (e *Enforcer) enforce(matcher string, explains *[]string, rvals ...interfac
...
+				rvals = rvals[1:]
+			}
+		default:
+			break
+		}
+	}
+	paramType := strings.TrimPrefix(rtype, "r")
+	mtype := "m" + paramType
+	ptype := "p" + paramType
 	var expString string
```
[the remark this answers is not in the document]
Comment (tangyang9464) [reply enforcer.go:483]: you are right.Here should match with "r" as the prefix

**Comment by hsluoyz** [comment]
@closetool plz review.

**Comment by tangyang9464** [comment]
There are some errors.I'll make a new PR

**Comment by hsluoyz** [comment]
Replaced by: https://github.com/casbin/casbin/pull/838

**Pull request closed without merging**